thirdparty_builtin/CMakeLists.txt

```diff
@@ -145,39 +145,40 @@ if(ENABLE_TESTS)
         endif()
     endif()
 
-    if(ENABLE_BENCHMARKS)
-        if(WIN32 AND BUILD_SHARED_LIBS)
-          message(FATAL_ERROR "Benchmarks cannot be built when BUILD_SHARED_LIBS=On")
-        endif()
+endif()
 
-        ## google benchmark support
-        add_subdirectory(gbenchmark-master-2017-05-19
-                         ${BLT_BUILD_DIR}/thirdparty_builtin/gbenchmark-master-2017-05-19)
+if(ENABLE_BENCHMARKS)
+    if(WIN32 AND BUILD_SHARED_LIBS)
+      message(FATAL_ERROR "Benchmarks cannot be built when BUILD_SHARED_LIBS=On")
+    endif()
 
-        if (UNIX AND NOT APPLE)
-          find_library(RT_LIBRARIES rt)
-        endif()
+    ## google benchmark support
+    add_subdirectory(gbenchmark-master-2017-05-19
+                     ${BLT_BUILD_DIR}/thirdparty_builtin/gbenchmark-master-2017-05-19)
 
-        blt_register_library(NAME gbenchmark
-                             INCLUDES ${benchmark_SOURCE_DIR}/include ${benchmark_SOURCE_DIR}
-                             LIBRARIES benchmark ${RT_LIBRARIES}
-                             TREAT_INCLUDES_AS_SYSTEM ON
-                             )
+    if (UNIX AND NOT APPLE)
+      find_library(RT_LIBRARIES rt)
+    endif()
 
-        list(APPEND _blt_tpl_targets benchmark)
+    blt_register_library(NAME gbenchmark
+                         INCLUDES ${benchmark_SOURCE_DIR}/include ${benchmark_SOURCE_DIR}
+                         LIBRARIES benchmark ${RT_LIBRARIES}
+                         TREAT_INCLUDES_AS_SYSTEM ON
+                         )
 
-        # This sets up a target to run the benchmarks
-        add_custom_target(run_benchmarks 
-                          COMMAND ctest -C Benchmark -VV -R benchmark
-                          WORKING_DIRECTORY ${PROJECT_BINARY_DIR}
-                          )
-    endif()
+    list(APPEND _blt_tpl_targets benchmark)
 
-    # Set the folder property of the blt thirdparty libraries 
-    if(ENABLE_FOLDERS)
-        foreach(tpl ${_blt_tpl_targets})
-            blt_set_target_folder(TARGET ${tpl} FOLDER blt/thirdparty)
-        endforeach()
-    endif()
+    # This sets up a target to run the benchmarks
+    add_custom_target(run_benchmarks 
+                      COMMAND ctest -C Benchmark -VV -R benchmark
+                      WORKING_DIRECTORY ${PROJECT_BINARY_DIR}
+                      )
+endif()
 
+# Set the folder property of the blt thirdparty libraries 
+if(ENABLE_FOLDERS)
+    foreach(tpl ${_blt_tpl_targets})
```

Comment:
With this code separated from the conditions, I think you will also need to move the line that initializes ``_blt_tpl_targets`` out of the ``ENABLE_TESTING`` part.

I.e. this line:
```cmake
 set(_blt_tpl_targets) # tracks names of enabled tpl targets
```
should be above the ``if(ENABLE_TESTING)```

See: https://github.com/LLNL/blt/pull/137/files#diff-f7bc02caeeac3bac796502ed63313eafR46

Reply:
👍